controllers/machineset.go

```diff
@@ -521,14 +554,9 @@ func (r *AppWrapperReconciler) removeLabelFromMachine(ctx context.Context, aw *a
 		klog.Infof("Error removing label from machine: %v", err)
 		return err
 	}
-	return nil
-}
 
-func (r *AppWrapperReconciler) removeLabelFromNode(ctx context.Context, aw *arbv1.AppWrapper, nodeName string) error {
-	labelPatch := fmt.Sprintf(`[{"op":"remove","path":"/metadata/labels/%s","value":"%s" }]`, aw.Name, aw.Name)
-	_, err := r.kubeClient.CoreV1().Nodes().Patch(ctx, nodeName, types.JSONPatchType, []byte(labelPatch), metav1.PatchOptions{})
-	if err != nil {
-		klog.Infof("Error deleted label to machines %v", err)
+	// Add Machine label
+	if err := r.addLabelToMachine(ctx, newAw, machineName); err != nil {
```

Comment:
The two patches can be collapsed into a single patch request. The `labelPatch` can be an array of operations, so the one from `addLabelToMachine` can be added to the one here, and the `addLabelToMachine` method removed.